fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -1099,6 +1099,59 @@ void randomlyRemoveAllRecords() throws IOException {
                     assertThat(partitionCounts, contains(0)));
     }
 
+    static Stream<Arguments> multiUpdate() {
+        return ParameterizedTestUtils.cartesianProduct(
+                ParameterizedTestUtils.booleans("isSynthetic"),
+                ParameterizedTestUtils.booleans("isGrouped"),
+                Stream.of(0, 10),
+                Stream.of(0, 1, 4),
+                Stream.of(5365));
+    }
+
+    @ParameterizedTest
+    @MethodSource("multiUpdate")
+    void multipleUpdatesInTransaction(boolean isSynthetic, boolean isGrouped, int highWatermark, int updateCount, long seed) throws IOException {
+        final int documentCount = 15;
```

Comment:
Would it make sense to have this be random, perhaps between 11 and 19?
As it is you either have 15 documents in the partition, or 10 and 5. Putting some randomness in there might find some interesting edge cases.